ouroboros-consensus/src/Ouroboros/Consensus/Mempool/Impl.hs

```diff
@@ -213,23 +238,23 @@ initVR cfg = \knownValid (tip, st) -> ValidationResult {
 extendVR :: ApplyTx blk
          => LedgerConfig blk
          -> Bool -- ^ Were these transactions previously validated?
-         -> GenTx blk
+         -> (GenTxId blk, GenTx blk)
          -> ValidationResult blk
          -> ValidationResult blk
-extendVR cfg prevApplied tx
+extendVR cfg prevApplied (txid, tx)
```

Comment:
This is an optimisation, asking for the txid to be passed in, rather than calculating it from the tx. So we rely on the txid and tx matching. It's entirely possible for callers to get this wrong. I'd add an assertion here that they do match.